### PR TITLE
Removes Junit and mockito dependecies

### LINKED ISSUE #2133: replace mockito inline by mockito core
lib is deprecated and causes issues with java 21 and up. Also update workflows to java 21. Example https://github.com/microsoft/kiota-java/pull/1538

### PR DESCRIPTION
**Pull request by timayabi2020**
Fixes #2133 
Removes Junit and Mockito dependencies because there are no unit tests for this SDK. This will also reduce the number of dependabot PRs

### CODE REVIEW

File: gradle/dependencies.gradle [hunk 1]
```diff
@@ -2,7 +2,7 @@ dependencies {
     // Use JUnit test framework
     testImplementation 'org.junit.jupiter:junit-jupiter-api:5.11.0'
     testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.11.0'
-    testImplementation 'org.mockito:mockito-inline:5.2.0'
+    testImplementation 'org.mockito:mockito-core:5.12.0'
```
Comment (Ndiritu) [inline gradle/dependencies.gradle:5]:
```suggestion
    testImplementation 'org.mockito:mockito-core:5.13.0'
```

**Comment by timayabi2020** [comment]
> Can you also bump the java version of the workflows to 21 please?

The versions in the workflows were already bumped to 21. The one that we will need to bump is in the java core. I also think we actually don't need mockito here because there are no unit tests here

**Comment by baywet** [comment]
since we don't have any unit test on the generated code we I think we could remove the junit and mockito dependencies, it'd reduce the number of dependabot PRs we get

**Comment by baywet** [comment]
@timayabi2020 I can see that you've requested a review but the code hasn't changed. Did you forget to push by any chance?

**Comment by timayabi2020** [comment]
> @timayabi2020 I can see that you've requested a review but the code hasn't changed. Did you forget to push by any chance?

I think my network timed out during the process

**Review by baywet — APPROVED** [review]

**Pull request merged**